internal/dag/builder.go

```diff
@@ -209,11 +209,14 @@ func (b *builder) lookupSecret(m meta) *Secret {
 	s := &Secret{
 		Object: sec,
 	}
-	if b.secrets == nil {
-		b.secrets = make(map[meta]*Secret)
+	if s.Cert() != nil && len(s.Cert()) > 0 && s.PrivateKey() != nil && len(s.PrivateKey()) > 0 {
```

Comment:
Could we move this to a helper method, would make it easier to test the logic and clean up the logic flow.

Reply:
Sure